core/client_test.go

```diff
@@ -0,0 +1,54 @@
+package core
+
+import (
+	"context"
+	"testing"
+
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClientLifecycle(t *testing.T) {
+	client := MockClient()
+	err := client.Stop()
+	require.NoError(t, err)
+}
+
+func TestClient_Status(t *testing.T) {
+	client := MockClient()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	status, err := client.Status(ctx)
+	require.Nil(t, err)
+	assert.NotNil(t, status)
+
+	err = client.Stop()
+	require.NoError(t, err)
+}
+
+func TestClient_StartBlockSubscription_And_GetBlock(t *testing.T) {
+	client := MockClient()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	eventChan, err := client.Subscribe(ctx, "NewBlock/Events", types.QueryForEvent(types.EventNewBlock).String())
+	require.Nil(t, err)
+
+	for i := 1; i <= 3; i++ {
+		<-eventChan
+		// check that `Block` works as intended (passing nil to get block at latest height)
+		block, err := client.Block(ctx, nil)
+		require.Nil(t, err)
+		require.Equal(t, int64(i), block.Block.Height)
+	}
+
+	// TODO(@Wondertan): For some reason, local client errors with no subscription found, when it exists.
```

Comment:
You must `Start` the Client before Subscribing.

Reply:
so `client.Start` then `client.Subscribe`

Reply:
If you would look closer to `StartMockNode` and `MockClient`, you will see that it returns a started Node. Also, if the node wasn't started, the code above wouldn't work. Therefore, your suggestion is incorrect.

Reply:
Is this still an issue?

Reply:
Not anymore

Reply:
https://github.com/celestiaorg/celestia-node/pull/55 reworks this.